README.rst
Analysis of Select Semiconductor Industry Companies
====================================================


.. image:: https://img.shields.io/badge/license-MIT-red.svg
    :target: LICENSE.txt
.. image:: https://img.shields.io/github/v/release/leeway64/Analysis_of_Select_Semiconductor_Industry_Companies
    :alt: GitHub release (latest by date)


Motivation
-----------

The semiconductor industry is perhaps the most important industry of the 21st century. It provides
the backbone to most of the advanced technology we almost always take for granted. Without
semiconductors, technology such as mobile phones, computers, and the internet would not be possible.

Big shout out to `The Economist <https://github.com/TheEconomist>`_. They always produce excellent
reporting and graphs/plots. For a particularly well-done project that they undertook, refer to their
`COVID-19 excess deaths model <https://github.com/TheEconomist/covid-19-the-economist-global-excess-deaths-model>`_


Abstract
----------

This project analyzes the relative strengths of some of the most important manufacturers in the 
semiconductor industry; the companies included in this study are Intel, TSMC (Taiwan Semiconductor
Manufacturing Company), Samsung Electronics, and SMIC (Semiconductor Manufacturing International
Corporation).

Read the full report/analysis `here <analysis/README.rst>`_.

.. image:: images/Stock_prices.png

.. image:: images/Plotted_statistics.png


Drawing the plots
-------------------

First, install this project onto your local machine and install the required packages in the R Console:

.. code-block:: bash

	git clone https://github.com/leeway64/Analysis_of_Select_Semiconductor_Industry_Companies.git
	R
	renv::restore()

Upon entering the R Console, you should see something to the effect of
``Project 'path/to/Analysis_of_Select_Semiconductor_Industry_Companies' loaded. [renv 0.16.0]``

Then, in the R Console, run the following:

.. code-block::

	source(here::here("scripts", "semiconductor_industry_companies_data_visualization.R"))
	

Alternatively, you can also run `semiconductor_industry_companies_data_visualization.R <scripts/semiconductor_industry_companies_data_visualization.R>`_ in RStudio.


References for Plots
-----------------------

This section serves as a list of the references used to create the plots. Refer to the
`report <./analysis/README.rst>`_ for a complete list of sources, including sources used to
create the report.

Intel
~~~~~~

**Revenue, R & D spending**: `Financial Results <https://www.intc.com/financial-info/financial-results>`_

**Stock price**: `Yahoo Finance <https://finance.yahoo.com/quote/INTC/history?period1=322099200&period2=1630713600&interval=1wk&filter=history&frequency=1wk&includeAdjustedClose=true>`_

TSMC
~~~~~

**Revenue, profit, R & D spending**: `Annual Reports <https://investor.tsmc.com/english/annual-reports>`_

**Stock price**: `Yahoo Finance <https://finance.yahoo.com/quote/TSM/history?period1=876355200&period2=1630713600&interval=1wk&filter=history&frequency=1wk&includeAdjustedClose=true>`_

Samsung
~~~~~~~~

**Profit**: `Financial Statements <https://www.samsung.com/global/ir/financial-information/audited-financial-statements/>`_

**Stock price**: `Yahoo Finance <https://finance.yahoo.com/quote/005930.KS/history?period1=946857600&period2=1630713600&interval=1wk&filter=history&frequency=1wk&includeAdjustedClose=true>`_

SMIC
~~~~~

**Profit**: `Financials <https://www.smics.com/en/site/company_financialSummary>`_

**Stock price**: `Yahoo Finance <https://finance.yahoo.com/quote/0981.HK/history?period1=1079481600&period2=1630713600&interval=1wk&filter=history&frequency=1wk&includeAdjustedClose=true>`_



Third Party Software
----------------------
- `renv <https://rstudio.github.io/renv/>`_ (MIT License): Reproducible environment creation library;

- `here <https://here.r-lib.org/>`_ (MIT License): Library for making file referencing easier.

- `rio <https://cran.r-project.org/web/packages/rio/index.html>`_ (GPL-2): File input/output.

- `dplyr <https://dplyr.tidyverse.org/>`_ (MIT License): Data manipulation library.

- `ggplot2 <https://ggplot2.tidyverse.org/>`_ (MIT License): Plotting library.

- `gridExtra <https://cran.r-project.org/web/packages/gridExtra/index.html>`_ (GPL-2 | GPL-3): Grid graphics manipulation.

- `box <https://klmr.me/box/>`_ (MIT License): Package to increase modularity of R code.

The ``grid`` library comes included with R.


License
----------

My `analysis <analysis/README.rst>`_ is licensed under the `Creative Commons Attribution 4.0 
International license <https://creativecommons.org/licenses/by/4.0/>`_.

The source code used to create the plots is licensed under the `MIT License <LICENSE-CODE.txt>`_.


CHANGELOG.md
# Changelog
All changes between versions for this project will be documented in this file.

This changelog format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and
this project follows [semantic versioning](https://semver.org/).


## Unreleased
- None


## [1.1.1] - 2023-7-3

### Changed
- Made small formatting edits to README

### Removed
- Deleted scripts/requirements.rst


## [1.1.0] - 2022-11-6

### Added
- Updated README and analysis
- Integrated box and renv with project
- Added CC-BY-4.0 License text


## [1.0.0] - 2022-4-18
- First release


scripts/semiconductor_industry_companies_data_visualization.R
# Visualizes several statistics of Intel, TSMC, Samsung Electronics, and SMIC.

library(here)
library(rio)
library(dplyr)
library(ggplot2)
library(grid)
library(gridExtra)


# Imports a data file from the "data" folder
read_data <- function(data_file){
    data <- import_list(here("data", data_file))  # Import all sheets in an Excel spreadsheet
    return(data)
}


# Plots the closing stock price of each week for each company
plot_stock_price <- function(){
    path <- "stock_price/"
    Intel <- read_data(paste(path, "INTC.xlsx", sep=""))$INTC
    TSMC <- read_data(paste(path, "TSM.xlsx", sep=""))$TSM
    
    Samsung <- read_data(paste(path, "005930.KS.xlsx", sep=""))$"005930.KS"
    # Convert Korean Republic won to US dollars
    KRW_to_USD <- Map({function (x) x*0.00086}, Samsung$Close)
    KRW_to_USD <- unlist(KRW_to_USD)
    Samsung$Close <- KRW_to_USD
    
    SMIC <- read_data(paste(path, "0981.HK.xlsx", sep=""))$"0981.HK"
    # Convert Hong Kong dollars to US dollars
    HKD_to_USD <- Map({function (x) x*0.13}, SMIC$Close)
    SMIC$Close <- unlist(HKD_to_USD)

    dev.new()
    print(ggplot()+
              geom_line(aes(x=Intel$Date, y=Intel$Close, color="Intel"))+
              geom_line(aes(x=TSMC$Date, y=TSMC$Close, color="TSMC"))+
              geom_line(aes(x=Samsung$Date, y=Samsung$Close, color="Samsung"))+
              geom_line(aes(x=SMIC$Date, y=SMIC$Close, color="SMIC"))+
              labs(title='Stock price vs. date', x='Date', y='Stock price (USD)'))
}


plot_revenue <- function(){
    revenue_data <- read_data("revenue/revenue.xlsx")
    Intel <- revenue_data$Intel
    # Convert millions of USD to thousands of USD
    millions_to_thousands <- Intel$Revenue %>% (function (x) x * 1000)
    Intel$Revenue <- unlist(millions_to_thousands)
    
    TSMC <- revenue_data$TSMC
    # Convert millions of New Taiwan dollars to thousands of US dollars
    NTD_to_USD <- Map({function (x) x*0.036*1000}, TSMC$Revenue)
    NTD_to_USD <- unlist(NTD_to_USD)
    TSMC$Revenue <- NTD_to_USD

    # From the sources provided, Samsung and SMIC don't provide net revenue data
    
    return(ggplot()+
            geom_line(aes(x=Intel$Year, y=Intel$Revenue, color='Intel'))+
            geom_line(aes(x=TSMC$Year, y=TSMC$Revenue, color='TSMC'))+
            labs(title='Net revenue vs. year', x='Year', y='Revenue (USD, \'000)'))
}


# Plots the gross profit of the companies
plot_profit <- function(){
    profit_data <- read_data("profit/profit.xlsx")
    
    # Intel doesn't provide gross profit, only gross margin, in the sources provided
    
    TSMC <- profit_data$TSMC
    # Convert millions of New Taiwan dollars to thousands of US dollars
    NTD_to_USD <- Map({function (x) x*0.036*1000}, TSMC$Gross_profit)
    NTD_to_USD <- unlist(NTD_to_USD)
    TSMC$Gross_profit <- NTD_to_USD
    
    Samsung <- profit_data$Samsung
    SMIC <- profit_data$SMIC
    
    return(ggplot()+
              geom_line(aes(x=TSMC$Year, y=TSMC$Gross_profit, color='TSMC'))+
              geom_line(aes(x=Samsung$Year, y=Samsung$Gross_profit, color='Samsung'))+
              geom_line(aes(x=SMIC$Year, y=SMIC$Gross_profit, color='SMIC'))+
              labs(title='Gross profit vs. year', x='Year', y='Gross profit (USD, \'000)'))
}


# Plots R & D spending of the 4 companies as a percentage of net revenue
plot_RD_spending <- function(){
    RD_spending_data <- read_data("RD_spending/RD_spending.xlsx")
    
    Intel <- RD_spending_data$Intel
    TSMC <- RD_spending_data$TSMC

    # From the sources provided, Samsung and SMIC don't provide net revenue data
    
    return(ggplot()+
              geom_line(aes(x=Intel$Year, y=Intel$RD, color='Intel'))+
              geom_line(aes(x=TSMC$Year, y=TSMC$RD, color='TSMC'))+
              labs(title='R & D spending vs. year',
                   x='Year', y='R & D spending (% of net revenue)'))
}


main <- function(){
    plot_stock_price()
    dev.new()
    grid.arrange(plot_revenue(), plot_profit(), plot_RD_spending(), nrow=3,
                 top=textGrob("Plotted statistics of Intel, TSMC, Samsung Electronics, and SMIC"))
}


# Similar to __name__ == "__main__" in Python
if (is.null(box::name()))
{
    main()
}
